### ISSUE TITLE
Wrong path resolve

### ISSUE THREAD

**Issue by btamas** [lasso-js/lasso-require#28]
A had a project that worked before but today it's broken because of some of your modifications.
I cannot find a problem but it can be reproducible easily with the following steps:
```
npm install intl lasso-cli
echo "require('intl');" > main.js
lasso --main main.js
```
It said something like this:
```
Error: File does not exist: /home/user/node_modules/intl/home/user/node_modules/intl/locale-data/complete.js
    at getPathInfo (/home/user/node_modules/lasso-cli/node_modules/lasso/node_modules/raptor-modules/util/lib/getPathInfo.js:63:15)
...
```

Versions:
- "intl" : "1.1.0",
- "lasso-cli": "1.3.0"

We are using lasso to require a lot of npm libraries, but we experienced this issue only with intl. I cannot find any specific problem with this project. It worked in the last week so maybe it is caused by a new modification. Could you check it for me?

Thanks in advance and if you need any other information please let me know.

**Comment by btamas** [comment]
I find the problem. If I install intl and remove browser tags from package.json it is working.

**Comment by patrick-steele-idem** [comment]
@btamas what does the `browser` field look like in your `package.json`?

**Comment by patrick-steele-idem** [comment]
I did some changes to better support the `browser` field. If you could provide more info to help me reproduce the problem you are seeing that would be great.

**Comment by patrick-steele-idem** [comment]
Oh, I see. They are doing something weird with the `browser` field in the `intl` package: https://github.com/andyearnshaw/Intl.js/blob/5ef6f6d384fe8919edcedafa4537b40a082e5323/package.json#L9-L12

```json
  "browser": {
    "./locale-data/complete": false,
    "./locale-data/complete.js": false
  }
```

I'll work on a fix real fast.

**Comment by btamas** [comment]
@patrick-steele-idem Sorry about the non proper description, but yes you find the problem.
Thanks for you to work on the fix.

**Comment by btamas** [comment]
What's up @patrick-steele-idem? Is there any problem or it is more complicate than you thought before? Can I help you somehow?

**Comment by patrick-steele-idem** [comment]
Hey @btamas, I have been working on fixing the problem every chance I get, but unfortunately it did turn out to be bigger than I thought. While I could have hacked together a solution, I found that changes to the related code ended up leading to other problems elsewhere. As a result, I am refactoring much of the code related to resolving module and handling browser overrides. It is taking much longer than expected because I want to get it right and not risk breaking existing code (again). I'm going to go ahead and back out the breaking change to at least unblock you, but even the old behavior wasn't quite right. I'll send an update as soon as I have published a new version with the changes reverted. Thanks for your patience.
